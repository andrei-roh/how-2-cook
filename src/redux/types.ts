export const SET_USER = 'SET_USER';
export const SET_LOADING = 'SET_LOADING';
export const SET_ERROR = 'SET_ERROR';
export const ADD_RECIPES_TO_LIST = 'ADD_RECIPES_TO_LIST';
export const UPDATE_RECIPES_LIST = 'UPDATE_RECIPES_LIST';
export const ADD_IMAGES_TO_LIST = 'ADD_IMAGES_TO_LIST';
export const SET_RECIPE_SEARCH_INPUT = 'SET_RECIPE_SEARCH_INPUT';
export const SET_MANAGE_RECIPES_SCROLL_SIZE = 'SET_MANAGE_RECIPES_SCROLL_SIZE';
export const SET_MANAGE_RECIPES_SCROLL_DIRECTION =
  'SET_MANAGE_RECIPES_SCROLL_DIRECTION';
export const SET_SHOW_MENU = 'SET_SHOW_MENU';
export const ADD_INGREDIENTS_TO_LIST = 'ADD_INGREDIENTS_TO_LIST';
export const SET_INGREDIENT_SEARCH_INPUT = 'SET_INGREDIENT_SEARCH_INPUT';
export const UPDATE_INGREDIENTS_LIST = 'UPDATE_INGREDIENTS_LIST';


