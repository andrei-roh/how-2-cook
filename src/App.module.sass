@use 'src/style/colors'

.app 
  padding-top: 46px
  background-color: colors.$white
  height: 100vh
  box-sizing: border-box
  display: flex
  flex-direction: column
  align-items: center

