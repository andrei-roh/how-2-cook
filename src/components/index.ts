export { Header } from './Header/Header';
export { Button } from './Button/Button';
export { Input } from './Input/Input';
export { Modal } from './Modal/Modal';
export { Notification } from './Notification/Notification';
export { ImageLoader } from './ImageLoader/ImageLoader';
export { TextArea } from './TextArea/TextArea';
export { Select } from './Select/Select';