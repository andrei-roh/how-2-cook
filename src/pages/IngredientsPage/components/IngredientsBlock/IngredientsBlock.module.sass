@use 'src/style/colors'
@import 'src/style/typography'

.ingredientsServiceWrapper
    width: 100%
    margin: calc(100vh / 4 - 100px) 0 24px 0

    .ingredientsBlockLogo
        margin-top: 180px
        height: 200px

    .emptyMessage
        @extend .T1
        text-align: center

.ingredientsBlockWrapper
    width: 100%
    margin-top: 150px
    padding-bottom: 16px
    justify-content: center
    align-items: center