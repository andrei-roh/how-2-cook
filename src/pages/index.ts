export { TitlePage } from './TitlePage/TitlePage';
export { RecipesPage } from './RecipesPage/RecipesPage';
export { CreateRecipePage } from './CreateRecipePage/CreateRecipePage';
export { ShowRecipePage } from './ShowRecipePage/ShowRecipePage';
export { EditRecipePage } from './EditRecipePage/EditRecipePage';
export { IngredientsPage } from './IngredientsPage/IngredientsPage';
export { WeekPlanningPage } from './WeekPlanningPage/WeekPlanningPage';
export { MyPlansPage } from './MyPlansPage/MyPlansPage';