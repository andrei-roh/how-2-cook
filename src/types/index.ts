import { Action } from '@reduxjs/toolkit';

export interface IUser {
  email: string;
}

export enum DishType {
  Main = 'main',
  Soup = 'soup',
  Dessert = 'dessert',
  Salat = 'salat',
  Snack = 'snack',
}

export type RecipeImage = {
  [key: string]: string;
};

export interface IIngredient {
  id: string;
  name: string;
}

export interface IRecipe {
  id: string;
  imageUrl: string;
  name: string;
  type: DishType;
  isVegan: boolean;
  ingredients: string[];
  description: string;
  composition?: string;
  createdAt?: string;
  createdBy?: string;
  updatedAt?: string;
  updatedBy?: string;
}

export interface IPlanningList {
  id: string;
  name: string;
  ingredientsList: string[];
  recipesList: string[];
}

export interface IState {
  user: IUser;
  loading: boolean;
  error: null | string;
  recipesList: IRecipe[];
  imagesList: RecipeImage[];
  recipesPage: {
    recipeSearchInput: string;
    scrollSize: number;
    scrollDirection: ScrollDirection;
  };
  isMenu: boolean;
  ingredientsList: IIngredient[];
  ingredientsPage: {
    ingredientSearchInput: string;
  }
  previousRoute: string;
  planningPage: {
    ingredientsList: string[];
    recipesList: IRecipe[];
  }
}

export interface IAction extends Action {
  type: string;
  payload: never;
}

export interface IValidationError {
  [key: string]: string;
}

export enum FirebaseErrorType {
  InvalidEmail = 'auth/invalid-email',
  InvalidPassword = 'auth/invalid-credential',
  TooManyRequsts = 'auth/too-many-requests',
}

export enum ImageType {
  Webp = 'image/webp',
  Jpeg = 'image/jpeg',
  Jpg = 'image/jpg',
  Png = 'image/png',
}

export interface IOption {
  name: string;
  value: string;
}

export enum ScrollDirection {
  Up = 'up',
  Down = 'down',
}

export enum Severity {
  Success = 'success',
  Info = 'info',
  Warning = 'warning',
  Error = 'error',
}

export interface ICurrentRecipe extends Partial<IRecipe> {}
