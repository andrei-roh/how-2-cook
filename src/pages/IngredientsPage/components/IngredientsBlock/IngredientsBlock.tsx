import css from './IngredientsBlock.module.sass';
import { IIngredient } from 'src/types';
import Fridge from 'src/assets/hand-drawn-food.svg';
import Pancakes from 'src/assets/pancakes.svg';
import Typography from '@mui/material/Typography';
import Stack from '@mui/material/Stack';
import { IngredientCard } from '../IngredientCard/IngredientCard';

interface IngredientsBlockProps {
  ingredients: IIngredient[];
  isSearch: boolean;
}

export const IngredientsBlock = ({
  ingredients,
  isSearch,
}: IngredientsBlockProps) => {
  if (ingredients.length === 0) {
    return (
      <Stack
        direction='column'
        useFlexGap
        spacing={2}
        className={css.ingredientsServiceWrapper}
      >
        {isSearch ? (
          <>
            <img className={css.ingredientsBlockLogo} src={Pancakes} />
            <Typography className={css.emptyMessage}>
              Ингредиенты не найдены
            </Typography>
          </>
        ) : (
          <>
            <img className={css.ingredientsBlockLogo} src={Fridge} />
            <Typography className={css.emptyMessage}>
              Добавьте первый ингредиент
            </Typography>
          </>
        )}
      </Stack>
    );
  }

  return (
    <Stack
      direction='column'
      useFlexGap
      spacing={2}
      className={css.ingredientsBlockWrapper}
    >
      {ingredients.map((ingredient) => (
        <IngredientCard key={ingredient.id} ingredient={ingredient} />
      ))}
    </Stack>
  );
};
