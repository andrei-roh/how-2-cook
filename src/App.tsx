import { Outlet } from 'react-router-dom';
import css from './App.module.sass';
import { Header, Notification } from './components';

const App = () => {
  return (
    <>
      <Header />
      <div className={css.app}>
        <Outlet />
      </div>
      <Notification />
    </>
  );
};

export default App;
