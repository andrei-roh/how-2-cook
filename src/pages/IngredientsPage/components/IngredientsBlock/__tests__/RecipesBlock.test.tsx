import { describe, expect, it, vi } from 'vitest';
import { render } from '@testing-library/react';
import { IngredientsBlock } from '../IngredientsBlock';
import { IIngredient, IUser } from 'src/types';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';

const navigateMock = vi.fn();
const dispatchMock = vi.fn();
const userMock = { email: 'Test@email.com' };
const recipesMock: IIngredient[] = [
  {
    id: '00000000-0000-0000-0000-000000000001',
    name: 'Test Ingredient 1',
  },
  {
    id: '00000000-0000-0000-0000-000000000002',
    name: 'Test Ingredient 2',
  },
  {
    id: '00000000-0000-0000-0000-000000000003',
    name: 'Test Ingredient 3',
  },
];

const stateMock = {
  user: userMock as IUser,
  loading: false,
  error: null,
};

const reducerMock = (state = stateMock) => {
  return state;
};

const storeMock = configureStore({ reducer: reducerMock });

vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');

  return {
    ...actual,
    useNavigate: () => ({ navigate: navigateMock }),
  };
});

vi.mock('react-redux', async () => {
  const actual = await vi.importActual('react-redux');

  return {
    ...actual,
    useDispatch: () => ({ dispatch: dispatchMock }),
  };
});

describe('Testing: IngredientsBlock', () => {
  it('should render', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <IngredientsBlock ingredients={recipesMock} isSearch={false} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render if there are no ingredients', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <IngredientsBlock ingredients={[]} isSearch={false} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render if have not found ingredient', () => {
    const { container } = render(
      <Provider store={storeMock}>
        <IngredientsBlock ingredients={[]} isSearch={true} />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
