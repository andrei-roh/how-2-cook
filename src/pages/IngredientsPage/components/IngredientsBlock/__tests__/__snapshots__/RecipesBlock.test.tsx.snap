// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing: IngredientsBlock > should render 1`] = `
<div>
  <div
    class="MuiStack-root _ingredientsBlockWrapper_092a35 css-1ucnu3w-MuiStack-root"
  >
    <div
      class="_ingredientContainer_dd3edf MuiBox-root css-0"
      id="00000000-0000-0000-0000-000000000001"
      name="Test Ingredient 1"
    >
      <span
        class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
      >
        Test Ingredient 1
      </span>
      <div
        class="_buttonsPanel_dd3edf MuiBox-root css-0"
      >
        <img
          alt="Edit Ingredient Button"
          class="_ingredientBlockLogo_dd3edf"
          src="/src/assets/pencil.svg"
        />
        <img
          alt="Delete Ingredient Button"
          class="_ingredientBlockLogo_dd3edf"
          src="/src/assets/paper-bin.svg"
        />
      </div>
    </div>
    <div
      class="_ingredientContainer_dd3edf MuiBox-root css-0"
      id="00000000-0000-0000-0000-000000000002"
      name="Test Ingredient 2"
    >
      <span
        class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
      >
        Test Ingredient 2
      </span>
      <div
        class="_buttonsPanel_dd3edf MuiBox-root css-0"
      >
        <img
          alt="Edit Ingredient Button"
          class="_ingredientBlockLogo_dd3edf"
          src="/src/assets/pencil.svg"
        />
        <img
          alt="Delete Ingredient Button"
          class="_ingredientBlockLogo_dd3edf"
          src="/src/assets/paper-bin.svg"
        />
      </div>
    </div>
    <div
      class="_ingredientContainer_dd3edf MuiBox-root css-0"
      id="00000000-0000-0000-0000-000000000003"
      name="Test Ingredient 3"
    >
      <span
        class="MuiTypography-root MuiTypography-button css-1f0on15-MuiTypography-root"
      >
        Test Ingredient 3
      </span>
      <div
        class="_buttonsPanel_dd3edf MuiBox-root css-0"
      >
        <img
          alt="Edit Ingredient Button"
          class="_ingredientBlockLogo_dd3edf"
          src="/src/assets/pencil.svg"
        />
        <img
          alt="Delete Ingredient Button"
          class="_ingredientBlockLogo_dd3edf"
          src="/src/assets/paper-bin.svg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Testing: IngredientsBlock > should render if have not found ingredient 1`] = `
<div>
  <div
    class="MuiStack-root _ingredientsServiceWrapper_092a35 css-1ucnu3w-MuiStack-root"
  >
    <img
      class="_ingredientsBlockLogo_092a35"
      src="/src/assets/pancakes.svg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 _emptyMessage_092a35 css-ahj2mt-MuiTypography-root"
    >
      Ингредиенты не найдены
    </p>
  </div>
</div>
`;

exports[`Testing: IngredientsBlock > should render if there are no ingredients 1`] = `
<div>
  <div
    class="MuiStack-root _ingredientsServiceWrapper_092a35 css-1ucnu3w-MuiStack-root"
  >
    <img
      class="_ingredientsBlockLogo_092a35"
      src="/src/assets/hand-drawn-food.svg"
    />
    <p
      class="MuiTypography-root MuiTypography-body1 _emptyMessage_092a35 css-ahj2mt-MuiTypography-root"
    >
      Добавьте первый ингредиент
    </p>
  </div>
</div>
`;
